### DISCUSSION TITLE
Mullvad Qubes Proxy VM problem

### DISCUSSION THREAD

**Discussion by github59173** [PrivSec-dev/privsec.dev#134] [General]
I have run into a interesting problem. I have done some testing with the setup guide you have provided and can't seem to get a upload speed test. speedtest.net and fast.com both report 0.00Mbps upload speed in ever test in the connect disposable vm proxing through the mullvad vpn vm but not in the mullvad vm itself. The mullvad vpn settings are stock except for the manually set protocol of wire-guard (openvpn doesn't seem to work for proxyvm in this setup) running MullvadVPN-2023.3_x86_64 on fedora 37 and 38. I have also tested with the new "Quantum-resistant tunnel" on and off with no change.  Speedtests show excellent download speeds but 0.00Mpbs upload.  The latency for upload test is still low at 25ms and the test completes but no upload speed is recorded. Makes sense why most websites are broken. I'm at a lose. I have never encountered this kind of problem before.

**Comment by TommyTran732** [comment]
Can you try with the instructions on the latest post? https://privsec.dev/posts/qubes/using-mullvad-vpn-on-qubes-os/ 

It works fine for me.
